mesonbuild/dependencies/base.py

```diff
@@ -518,6 +525,23 @@ def get_configtool_variable(self, variable_name):
     def log_tried(self):
         return self.type_name
 
+    def get_variable(self, *, cmake: typing.Optional[str] = None, pkgconfig: typing.Optional[str] = None,
+                     configtool: typing.Optional[str] = None, default: typing.Optional[str] = None,
+                     pkgconfig_define: typing.Optional[typing.List[str]] = None) -> typing.Union[str, typing.List[str]]:
+        if configtool:
+            try:
+                # in the not required case this will be '' (empty string) if
+                # the variable is not found, in that case we want to continue
+                # and not return ''
```

Comment:
This seems a bit unfortunate as it would preclude the value from being an empty string in this case?

Reply:
You're right, I've changed it to force an error in that case so we can differentiate the two.